crates/iota-names/src/domain.rs

```diff
@@ -226,10 +226,12 @@ pub fn validate_label(label: &str) -> Result<&str, IotaNamesError> {
     for (i, character) in bytes.iter().enumerate() {
         match character {
             b'a'..=b'z' | b'0'..=b'9' => continue,
-            b'-' if i == 0 || i == len - 1 => {
```

Comment:
Not fully sure to see why it was failing tbh

Reply:
Because it was falling into

```rs
_ => return Err(IotaNamesError::InvalidLabelChar),
```

Reply:
Another way to put it:
```rs
match character {
      b'a'..=b'z' | b'0'..=b'9' => continue,
      b'-' if i > 0 || i < len - 1 => continue,
      b'-' => return Err(IotaNamesError::HyphensAsFirstOrLastLabelChar),
      _ => return Err(IotaNamesError::InvalidLabelChar((*character) as char, i)),
  };
```

Reply:
Ah yeah, obviously, thanks @marc2332. Time for my morning coffee ☕